update constraints & constraintemplates to v1

tillepille opened the pull request:
Since the [3.6 release](https://open-policy-agent.github.io/gatekeeper/website/docs/constrainttemplates/#v1-constraint-template) of Gatekeeper, the ConstraintTemplate have apiVersion v1.

See this more a proposal than a perfect solution, probably.. 
I'm definetly up for feedback!

This targets #205

jalseth reviewed (commented):
Hi @tillepille thanks for the PR! Older versions of Gatekeeper do not support this version of the CRD, so if we make this change we will break Konstraint for everyone using older versions. I think we should implement this behind a flag allowing the user to choose, and should default to the old v1beta1 version of the CRD until Gatekeeper removes support for that version of the CRD, at which point we would make v1 default and keep v1beta1 for a while for compatibility. What do you think about that?

tillepille commented:
@jalseth I added a flag, so that the default stays on v1beta1 but `konstraint create --generate-v1` will generate v1 manifests.

jalseth commented on internal/commands/create.go:
```diff
@@ -58,6 +63,7 @@ Create constraints with the Gatekeeper enforcement action set to dryrun
 	cmd.PersistentFlags().StringP("output", "o", "", "Specify an output directory for the Gatekeeper resources")
 	cmd.PersistentFlags().BoolP("dryrun", "d", false, "Sets the enforcement action of the constraints to dryrun, overriding the @enforcement tag")
 	cmd.PersistentFlags().Bool("skip-constraints", false, "Skip generation of constraints")
+	cmd.PersistentFlags().Bool("generate-v1", false, "Generate apiVersionv1 objects")
```
nit: I think this flag is a bit ambiguous. It would be more clear if this was a string flag along the lines of `constraint-template-version` with a default of `v1beta1` and an updated help text to match.

tillepille replied:
Done like that

jalseth commented on internal/commands/create.go:
```diff
@@ -10,11 +10,12 @@ import (
 
 	"github.com/ghodss/yaml"
-	"github.com/open-policy-agent/frameworks/constraint/pkg/apis/templates/v1beta1"
+	v1 "github.com/open-policy-agent/frameworks/constraint/pkg/apis/templates/v1"
+	v1beta1 "github.com/open-policy-agent/frameworks/constraint/pkg/apis/templates/v1beta1"
```
nit: Aliasing the import here doesn't do anything, as the name selected is the same as the original package.

tillepille replied:
removed it for v1beta1 but gofmt defaults to the `v1`.

jalseth commented on examples/container-deny-added-caps/template.yaml:
```diff
@@ -1,4 +1,4 @@
-apiVersion: templates.gatekeeper.sh/v1beta1
+apiVersion: templates.gatekeeper.sh/v1
```
Here and throughout the examples: We're still defaulting to v1beta1, so it doesn't make sense to change this to v1.

tillepille replied:
fixed.

tillepille commented:
Thanks @jalseth for your feedback!
I adjusted the flag. Seems a much better idea to make it future proof.
Also reverted the examples, this was a left over.

jalseth reviewed (approved):
Thanks @tillepille! LGTM so please squash the commits here with a descriptive commit message of the change and I will merge.
